src/excel_loader.py

```diff
@@ -0,0 +1,236 @@
+
+            # Append the new data to the end of the sheet
+            for row in dataframe_to_rows(selected_data, index=False, header=False):
+                sheet.append(row)
+
+            # Define a custom style for datetime values
+            time_style = NamedStyle(name="time_style", number_format="HH:MM")
+
+            # Formatting date time
+            for row in sheet.iter_rows(min_row=sheet.max_row - len(selected_data) + 1, max_row=sheet.max_row):
+                for cell in row:
+                    source_cell = df.iloc[cell.row - sheet.max_row + len(selected_data) - 1, cell.column - 1]
+
+                     # Check if the value is a datetime object and apply the custom style
+                    if isinstance(source_cell, pd.Timestamp):
+                        cell.style = time_style
+                    
+                    #preserve the row height
+                    sheet.row_dimensions[cell.row].height = sheet.row_dimensions[cell.row - sheet.max_row + len(selected_data)].height
+
+            # Preserve column width
+            for column_cells in sheet.columns:
+                length = max(len(str(cell.value)) for cell in column_cells)
+                sheet.column_dimensions[column_cells[0].column_letter].width = length + 2  # Additional padding
+
+            output_wb.save(output_file_name)
+
+        else:
+            print("Day not found in study schedule")
+        
+        append_empty_rows(sheet, 4)
+
+        i=i+1
+
+        print(f"Successfully created Daily_schedules.xslx")
+
+        excel_to_json(output_file_name, output_file_directory)
+
+
+def extract_name_from_schedule(task_name, current_study):
+    if 'Schedule' in task_name:
+        return task_name.split('Schedule')[0].strip()
+    else:
+        return current_study
+
+
+def assign_category(task_name):
+    # Enumerate task names based on keywords
+
+    doctor = ["EXAM", "EXAMINATION", "PHYSICAL", "DOCTOR", "ELIGIBILITY"]
+    nurse = ["DOSE", "PREDOSE", "AE", "ADVERSE", "MEDICATION", "ASSESS", "SITE", "ACCOUNTABILITY", "DISPENSE", "REPORTED", "INFUSION", "MONITOR"]
+    crt = ["BREAKFAST", "LUNCH", "DINNER", "SNACK", "SUPINE", "ECG", "VITAL", "VITALS", "ASSESSMENT", "COMMENCE", "BLOCK", "PREG", "DOA", "URINALYSIS", "URINE", "MSU", "WEIGHT", "BMI", "ALCOHOL", "CLEAN", "LINEN", "WALK"]
+    any = ["ADMIT", "WRIST", "WRISTBAND", "COMPLIANCE", "REMIND", "ENSURE", "RULES", "TOUR", "WORKBOOK", "DISCHARGE", "ENCOURAGE", "ORIENTATION", "RESTRICTIONS"]
+    phlebotomy = ["CANNULA", "CANNULATION", "BLOOD", "SAFETIES", "FLUSH"]
+    spiro = ["SPIROMETRY"]
+    sputum = ["SPUTUM"]
+    triplicate = ["TRIPLICATE"]
+    phone = ["PHONE"]
+    pharmacy = ["RANDOMISATION"]
+    date = ["DATE"]
+    day = ["DAY"]
+
+    assign_tasks = {'doctor' : doctor, 'nurse' : nurse, 'crt' : crt, 'any' : any, 'phlebotomy' : phlebotomy, 'spiro' : spiro, 'sputum' : sputum, 'triplicate' : triplicate, 'phone' : phone, 'pharmacy' : pharmacy, 'date' : date, 'day' : day}
```

Comment:
You could inline these definitions (and maybe move them to another file, or the top of this one so they don't clutter the function definition:

```python
assign_tasks = {
    "doctor": ["EXAM", "EXAMINATION", "PHYSICAL", "DOCTOR", "ELIGIBILITY"],
    "nurse": [
        "DOSE",
        "PREDOSE",
        "AE",
        "ADVERSE",
        "MEDICATION",
        "ASSESS",
        "SITE",
        "ACCOUNTABILITY",
        "DISPENSE",
        "REPORTED",
        "INFUSION",
        "MONITOR",
    ],
    "crt": [
        "BREAKFAST",
        "LUNCH",
        "DINNER",
        "SNACK",
        "SUPINE",
        "ECG",
        "VITAL",
        "VITALS",
        "ASSESSMENT",
        "COMMENCE",
        "BLOCK",
        "PREG",
        "DOA",
        "URINALYSIS",
        "URINE",
        "MSU",
        "WEIGHT",
        "BMI",
        "ALCOHOL",
        "CLEAN",
        "LINEN",
        "WALK",
    ],
    "any": [
        "ADMIT",
        "WRIST",
        "WRISTBAND",
        "COMPLIANCE",
        "REMIND",
        "ENSURE",
        "RULES",
        "TOUR",
        "WORKBOOK",
        "DISCHARGE",
        "ENCOURAGE",
        "ORIENTATION",
        "RESTRICTIONS",
    ],
    "phlebotomy": ["CANNULA", "CANNULATION", "BLOOD", "SAFETIES", "FLUSH"],
    "spiro": ["SPIROMETRY"],
    "sputum": ["SPUTUM"],
    "triplicate": ["TRIPLICATE"],
    "phone": ["PHONE"],
    "pharmacy": ["RANDOMISATION"],
    "date": ["DATE"],
    "day": ["DAY"],
}
```

Reply:
You could also use the enum values as the dictionary keys